packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -660,43 +666,60 @@ class TextSelectionOverlay {
       offset: lineAtOffset.baseOffset,
     );
 
-    final Rect lineBoundaries = Rect.fromPoints(
+    final Rect localLineBoundaries = Rect.fromPoints(
       renderEditable.getLocalRectForCaret(positionAtBeginningOfLine).topCenter,
       renderEditable.getLocalRectForCaret(positionAtEndOfLine).bottomCenter,
     );
+    final Rect globalLineBoundaries = Rect.fromPoints(
```

Comment:
Consider storing the transform and applying it instead of using `localToGlobal`.

Reply:
See my response to https://github.com/flutter/flutter/pull/146019#discussion_r1548802746.